sktm/__init__.py

```diff
@@ -25,6 +25,24 @@
 import sktm.patchwork
 
 
+def join_with_slash(base, *suffix):
+    """
+    Join parts of URL or path by slashes
+
+    Args:
+        base:    Base URL or path.
+        *suffix: Array of suffixes
+
+    Returns:
+           The URL or path string
+    """
+    for arg in suffix:
+        base = base.rstrip('/')
+        arg = arg.lstrip('/')
+        base = "{}/{}".format(base, arg)
+    return base
```

Comment:
Can I suggest a similar implementation `join_with_slash_another_imp` but easier to read and a little bit better in performance?

```python
In [81]: def join_with_slash(base, *suffix):
    ...:     """
    ...:     Join parts of URL or path by slashes
    ...: 
    ...:     Args:
    ...:         base:    Base URL or path.
    ...:         *suffix: Array of suffixes
    ...: 
    ...:     Returns:
    ...:            The URL or path string
    ...:     """
    ...:     for arg in suffix:
    ...:         base = base.rstrip('/')
    ...:         arg = arg.lstrip('/')
    ...:         base = "{}/{}".format(base, arg)
    ...:     return base
    ...: 

In [82]: 

In [82]: def join_with_slash_another_imp(base, *suffix):
    ...:     """
    ...:     Join parts of URL or path by slashes
    ...: 
    ...:     Args:
    ...:         base:    Base URL or path.
    ...:         *suffix: Array of suffixes
    ...: 
    ...:     Returns:
    ...:            The URL or path string
    ...:     """
    ...:     base = base.rstrip('/')
    ...:     url = [base]
    ...:     for arg in suffix:
    ...:         url.append(arg.strip('/'))
    ...:     return '/'.join(url)
    ...: 

In [83]: 

In [83]: join_with_slash('http://skt.com/', 'rh', '/foo/', 'bar/', '/baz')
Out[83]: 'http://skt.com/rh/foo/bar/baz'

In [84]: join_with_slash_another_imp('http://skt.com/', 'rh', '/foo/', 'bar/', '/baz')
Out[84]: 'http://skt.com/rh/foo/bar/baz'

In [85]: %timeit -q -o join_with_slash('http://skt.com/', 'rh', '/foo/', 'bar/', '/baz')
Out[85]: <TimeitResult : 1000000 loops, best of 3: 1.42 µs per loop>

In [86]: %timeit -q -o join_with_slash_another_imp('http://skt.com/', 'rh', '/foo/', 'bar/', '/baz')
Out[86]: <TimeitResult : 1000000 loops, best of 3: 923 ns per loop>```

Reply:
Nice to see you back @agustinhenze ! I love the performance improvement, but the implementation has the same issue as the first one from Hai did: if the last argument contains a trailing slash, it should stay there. So `join_with_slash_another_imp(baseurl, 'xmlrpc/')` should return `<baseurl>/xmlrpc/`, not  `<baseurl>/xmlrpc`. Do you think it's easy to modify your implementation to keep the performance improvement with this functionality?

Reply:
Yeah, I'm trying to sync... A lot of changes :smile:! Ok, you're right it doesn't keep the ending slash if it is there... Keeping the same implementation and just adding the corner case it would be something like

```python
In [87]: def join_with_slash_another_imp(base, *suffix):
    ...:     """
    ...:     Join parts of URL or path by slashes
    ...: 
    ...:     Args:
    ...:         base:    Base URL or path.
    ...:         *suffix: Array of suffixes
    ...: 
    ...:     Returns:
    ...:            The URL or path string
    ...:     """
    ...:     base = base.rstrip('/')
    ...:     url = [base]
    ...:     for arg in suffix:
    ...:         url.append(arg.strip('/'))
    ...:     ending = '/' if arg.endswith('/') else ''
    ...:     return '/'.join(url) + ending
```
And the performance still being quite better
```python
In [88]: %timeit -q -o join_with_slash_another_imp('http://skt.com/', 'rh', '/foo/', 'bar/', '/baz')
Out[88]: <TimeitResult : 1000000 loops, best of 3: 1.04 µs per loop>
```

Reply:
Looks good! /me always loves seeing performance improvements :) @danghai, can you use Augustin's implementation (and exchange the name of `url` variable to a more generic one so it's not confusing) and rebase please?